_posts/2024-01-03-opensearch-performance-improvements.md

```diff
@@ -0,0 +1,1151 @@
+
+* [**Concurrent segment search**](https://opensearch.org/blog/concurrent_segment_search/): Enabling every shard-level request to concurrently search across segments during the query phase resulted in latency reduction across multiple query types. Aggregate queries showed a 50%--70% improvement, range queries showed a 65% improvement, and aggregation queries with hourly data aggregations showed a 50% improvement.
+
+## Future roadmap
+
+The OpenSearch Project remains steadfast in its commitment to continuously improving the core engine performance in search, ingestion, and storage operations. [The OpenSearch Project roadmap on GitHub](https://github.com/orgs/opensearch-project/projects/153/views/1) is constantly evolving, and we are excited to share it with you. This roadmap places a special emphasis on the core engine advancements while also encompassing critical areas like benchmarking and query visibility. As part of our ongoing commitment, we plan to consistently update this roadmap with both short- and long-term improvement plans. We're keeping performance excellence at the forefront of our investments, and OpenSearch users can anticipate a series of impactful improvements in new releases in 2024, starting with OpenSearch 2.12.
+
+In the upcoming releases, we will continue to improve the core engine by targeting specific query types. We will also undertake broad strategic initiatives to further enhance the core engine through Protobuf integration, query rewrites, tiered caching, and SIMD and RUST implementations. In addition to improving the core engine, we are committed to improving OpenSearch tooling capabilities. One such improvement that we're currently working on is the query insights functionality, which helps identify the top N queries that impact performance. Additionally, OpenSearch is working on making benchmarks easier for community members to use. For a comprehensive list of investments and additional improvements, or to provide feedback, please check out the OpenSearch [Performance Roadmap](https://github.com/orgs/opensearch-project/projects/153/views/1) on GitHub.
+
+This concludes the main summary of OpenSearch performance improvements to date. The following appendix sections provide the benchmarking details for readers interested in replicating any run.
+
+* * *
+
+## Appendix: Detailed execution and results
+
+If you're interested in the details of the performance benchmarks we used, exploring the methodologies behind their execution, or examining the comprehensive results, keep reading. For OpenSearch users interested in establishing benchmarks and replicating these runs, we've provided comprehensive setup details alongside each result. This section provides the core engine performance comparison between the latest OpenSearch version (OpenSearch 2.11) and the state just before the OpenSearch fork, Elasticsearch 7.10.2, with a mid-point performance measurement on OpenSearch 2.3. 
```

Comment:
Can we add a line at the end of this paragraph:

"Also, we've identified items in the [performance roadmap](https://github.com/orgs/opensearch-project/projects/153/views/1) that require active enhancements due to observed regressions in specific areas."

Reply:
ohh looks like its merged! 👍